### ISSUE TITLE
Spurious unresolved import error for dependencies from old editions with keywords in the path

### ISSUE THREAD

**Issue by pmsanford** [rust-analyzer/rust-analyzer#6417]
I tried searching for another issue that seemed similar but the relevant keywords are.. keywords so there's lots of noise in that search. Apologies if this is known but I missed it.

If one of your dependencies is not `edition = 2018` and uses keywords that were _not_ keywords then but _are_ keywords now (for example, `async`) in the path, r-a will flag them as an unresolved import even though the project builds successfully. This even occurs if the path containing a keyword is private. I have a [minimal repro I posted here](https://github.com/pmsanford/rust-analyzer-keyword-repro).

The basic outline is having a dependency with a lib.rs similar to:
```
mod async;

pub use self::async::some_func;
pub fn other_func() -> i32 { 6 }
```

then, in your code:
```
use old_dep::other_func; // This is fine, no errors
use old_dep::some_func; // Redlined as "unresolved import" despite compiling fine
```

I ran into this in the wild with Termion; Termion is still not `edition = 2018` and [the async_stdin function](https://docs.rs/termion/1.5.5/termion/fn.async_stdin.html) is `pub use`d from a private module called `async`, causing [this line](https://github.com/pmsanford/simple-tui-rs/blob/master/src/main.rs#L5) to exhibit the problem for that function.

**Comment by bjorn3** [comment]
Only the 2018 edition is currently supported.

**Comment by pmsanford** [comment]
It was a bit tricky to figure out that that's the reason this one import wasn't being detected properly because my app _is_ 2018 and I had no idea Termion wasn't; I spent a nontrivial amount of time before I realized it would compile despite the error from rust-analyzer.

I suppose there's no clean fix here, though, without introducing some 2015 support, so that makes sense from a prioritization perspective. Happy to close the issue if 2015 support isn't on the near term roadmap.
